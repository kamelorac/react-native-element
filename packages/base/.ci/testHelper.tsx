import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
