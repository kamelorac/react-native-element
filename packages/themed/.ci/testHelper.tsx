import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
