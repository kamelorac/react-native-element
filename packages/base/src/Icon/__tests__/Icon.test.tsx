import React from 'react';
import { Icon } from '../index';
import { renderWithWrapper, fireEvent } from '../../../.ci/testHelper';

describe('Icon component', () => {
  it('should render with icon type', () => {
    const { toJSON } = renderWithWrapper(
      <Icon
        name="alert"
        type="octicon"
        reverse
        color="red"
        iconStyle={{ backgroundColor: 'peru', borderRadius: 30 }}
        onPress={jest.fn()}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should have onPress event', () => {
    const onPress = jest.fn();
    const { wrapper } = renderWithWrapper(
      <Icon onPress={onPress} name="wifi" />,
      'RNE__ICON__CONTAINER_ACTION'
    );
    fireEvent.press(wrapper);
    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('should apply default disabled styles', () => {
    const onPress = jest.fn();
    const { wrapper } = renderWithWrapper(
      <Icon onPress={onPress} name="edit" disabled />,
      'RNE__ICON'
    );
    expect(wrapper.props.style).toMatchObject({
      backgroundColor: '#D1D5D8',
    });
  });

  it('should apply custom disabled styles', () => {
    const onPress = jest.fn();
    const { wrapper } = renderWithWrapper(
      <Icon
        onPress={onPress}
        name="wifi"
        disabled
        disabledStyle={{ backgroundColor: 'pink' }}
      />,
      'RNE__ICON'
    );
    expect(wrapper.props.style).toMatchObject({
      backgroundColor: 'pink',
    });
  });

  it('should apply container style', () => {
    const { wrapper } = renderWithWrapper(
      <Icon name="wifi" containerStyle={{ backgroundColor: 'blue' }} />,
      'RNE__ICON__CONTAINER'
    );
    expect(wrapper.props.style).toMatchObject({
      backgroundColor: 'blue',
    });
  });

  it('should apply reverse styles', () => {
    const component = renderWithWrapper(<Icon name="wifi" reverse />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should set underlayColor to color when styles when underlayColor absent', () => {
    const onPress = jest.fn();
    const { toJSON } = renderWithWrapper(
      <Icon name="wifi" underlayColor={null} onPress={onPress} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should apply raised styles', () => {
    const component = renderWithWrapper(<Icon name="wifi" raised />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should have overridable testID', () => {
    const { queryByTestId } = renderWithWrapper(
      <Icon name="wifi" testID="wifiIcon" />
    );
    expect(queryByTestId('wifiIcon')).toBeTruthy();
  });
});
