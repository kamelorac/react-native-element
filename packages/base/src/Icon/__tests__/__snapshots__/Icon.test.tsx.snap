// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should apply raised styles 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "borderRadius": 28,
        "height": 52,
        "margin": 7,
        "shadowColor": "rgba(0,0,0, .4)",
        "shadowOffset": {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
        "width": 52,
      }
    }
    testID="RNE__ICON__CONTAINER"
  >
    <View
      accessibilityRole="button"
      android_ripple={
        {
          "borderless": false,
          "color": "rgba(0, 0, 0, 0.3)",
          "radius": -5,
        }
      }
      disabled={false}
      testID="RNE__ICON__CONTAINER_ACTION"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 28,
            "height": 52,
            "justifyContent": "center",
            "width": 52,
          }
        }
        testID="RNE__ICON"
      >
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "color": "#242424",
                "fontSize": 24,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
          testID="RNE__ICON__Component"
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Icon component should apply reverse styles 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "borderRadius": 28,
        "height": 52,
        "margin": 7,
        "overflow": "hidden",
        "width": 52,
      }
    }
    testID="RNE__ICON__CONTAINER"
  >
    <View
      accessibilityRole="button"
      android_ripple={
        {
          "borderless": false,
          "color": "rgba(36, 36, 36, 0.3)",
          "radius": -5,
        }
      }
      disabled={false}
      testID="RNE__ICON__CONTAINER_ACTION"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#242424",
            "borderRadius": 28,
            "height": 52,
            "justifyContent": "center",
            "width": 52,
          }
        }
        testID="RNE__ICON"
      >
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "color": "#ffffff",
                "fontSize": 24,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
          testID="RNE__ICON__Component"
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Icon component should render with icon type 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "borderRadius": 30,
        "height": 52,
        "margin": 7,
        "overflow": "hidden",
        "width": 52,
      }
    }
    testID="RNE__ICON__CONTAINER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="RNE__ICON__CONTAINER_ACTION"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": 28,
            "height": 52,
            "justifyContent": "center",
            "width": 52,
          }
        }
        testID="RNE__ICON"
      >
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "color": "#ffffff",
                "fontSize": 24,
              },
              {
                "backgroundColor": "peru",
                "borderRadius": 30,
              },
              {
                "fontFamily": "Octicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
          testID="RNE__ICON__Component"
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Icon component should set underlayColor to color when styles when underlayColor absent 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "overflow": "hidden",
      }
    }
    testID="RNE__ICON__CONTAINER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="RNE__ICON__CONTAINER_ACTION"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
        testID="RNE__ICON"
      >
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "color": "#242424",
                "fontSize": 24,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
          testID="RNE__ICON__Component"
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
